CONTRIBUTING.md

```diff
@@ -1,129 +1,224 @@
+this discards the old commits on your remote, and git won't take potentially
+destructive actions without confirmation.
+
+## Commit message guidelines
+
+When committing changes, we ask that you follow some guidelines to keep the
+history readable and consistent:
 
 * Start the message capitalized (only the first character must be in uppercase)
-* Be short and concise, the whole concept must fit one line
-* If a command is inlined, use backticks
-* Add a double-hashtag if the change matters for the changelog (See below)
-* If the commit fixes a bug start with 'Fix #number - '
-* For extra details, add an empty line and use asterisk item list below
-* Use present simple grammar tense (Add vs Added, Fix vs Fixed/Fixes)
-
-### Commit message hashtag list:
-
-* ##anal     - analysis related
-* ##asm      - assembler
-* ##bin      - binary parsing
-* ##build    - build fixes/changes
-* ##config   - config var changes/additions/renamings
-* ##cons     - console/terminal-related
-* ##crypto   - cryptography
-* ##debug    - debugger stuff
-* ##diff     - diffing code, strings, basic blocks, ...
-* ##disasm   - disassembler
-* ##doc      - documentation
-* ##egg      - the `r_lang` compiler
-* ##emu      - emulation, including esil
-* ##graph    - basic block graph, callgraph, ...
-* ##io       - related to the `r_io` library
-* ##json     - json fixes/changes
-* ##lang     - bindings
-* ##meta     - metadata handling other than printing
-* ##optimization-space/time optimizations
-* ##port     - portability (new OS/archs)
-* ##print    - printing data, structures, strings, tables, types ..
-* ##projects - saving/loading state
-* ##refactor - improve code quality
-* ##remote   - r2 over tcp, http, rap, serial .. including collaboration
-* ##search   - rafind2, / command, ..
-* ##shell    - command-line, argument parsing, new commands, ..
-* ##signatures-searching/generating them
-* ##test     - testing infrastructure fixes/changes
-* ##tools    - r2pm, rarun2, rax2 ... that don't fit in other categories
-* ##util     - core apis
-* ##visual   - visual ui, including panels
+* Be concise. A descriptive message under 100 characters is preferred, but may
+  not be possible in all situations. For large commits, it is acceptable to use
+  a summary line, followed by an empty line, then an asterisk item list of
+  changes.
+* If a command is inlined, use backticks, e.g.:
+
+```sh
+git commit -m 'Modify output of `ls`'
+```
+
+* Add a tag if the change falls into a relevant category (see below)
+* If the commit fixes an issue, you may optionally start the message with
+  `Fix #number - `
+* Use present simple tense and avoid past tense. Use "add", "fix", or "change"
+  instead of "added", "fixed", or "changed".
+
+### Commit message tag list
+
+| Tag              | Relevant changes |
+|------------------|------------------|
+| `##anal`         | Analysis |
+| `##asm`          | Assembly |
```

Comment:
somehow i tend to use ##arch ##disasm or #platform too. ##asm is usually tied to assembler capabilities only